src/v2/controllers/vreplicaset_controller/trusted/liveness_theorem.rs

```diff
@@ -33,18 +33,63 @@ pub open spec fn owned_selector_match_is(vrs: VReplicaSetView, obj: DynamicObjec
     &&& obj.metadata.deletion_timestamp.is_None()
 }
 
-// TODO: the current not_interfered_by invariant is radically strong. Weaken it later.
 pub open spec fn vrs_not_interfered_by(other_id: int) -> StatePred<ClusterState> {
     |s: ClusterState| {
         forall |msg| {
             &&& #[trigger] s.in_flight().contains(msg)
             &&& msg.content.is_APIRequest()
             &&& msg.src == HostId::Controller(other_id)
         } ==> match msg.content.get_APIRequest_0() {
-            APIRequest::CreateRequest(req) => req.obj.kind != Kind::PodKind,
-            APIRequest::UpdateRequest(req) => req.obj.kind != Kind::PodKind,
-            APIRequest::UpdateStatusRequest(req) => req.obj.kind != Kind::PodKind,
-            APIRequest::DeleteRequest(req) => req.key.kind != Kind::PodKind,
+            // Other controllers don't create pods owned by a VReplicaSet.
+            APIRequest::CreateRequest(req) => !{
+                let owner_references = req.obj.metadata.owner_references.get_Some_0();
+                &&& req.obj.kind == Kind::PodKind
+                &&& req.obj.metadata.namespace.is_Some()
```

Comment:
@codyjrivera do we need to require that requests from other controllers must carry a namespace? (tho they should)

Reply:
Not needed; thanks